src/pages/Trades.js

```diff
@@ -0,0 +1,193 @@
+import EditTradeModal from "../components/Trades/EditTradeModal.js";
+
+const TRADES_KEY = "trades";
+
+const Trades = () => {
+  const [user, setUser] = useState("noUser");
+  const [sort, setSort] = useState("timeA");
+  const [filter, setFilter] = useState("none");
+  const [filterCat, setFilterCat] = useState("none");
+  const [showModal, setShowModal] = useState(false);
+  const [showEditModal, setShowEditModal] = useState(false);
+  const [editTrade, setEditTrade] = useState(false);
+  const [tradesArr, setTradesArr] = useState([]);
+  const [tradelines, setTradelines] = useState(<div></div>);
+  const [masterTradesArr, setMasterTradesArr] = useState([]);
+  const [isDark, setIsDark] = useState(false);
+
+  useEffect(() => {
+    const unsubscribe = auth.onAuthStateChanged(() => {
+      if (auth.currentUser === null) {
+        setUser("noUser");
+      } else {
+        setUser(auth.currentUser.uid);
+      }
+    });
+    return () => {
+      unsubscribe();
+    };
+  });
+
+  useEffect(() => {
+    let loadedTrades = [];
+    const tradesRef = ref(database, `${user}/${TRADES_KEY}`);
+
+    const unsubscribe = onChildAdded(tradesRef, (data) => {
+      loadedTrades = [...loadedTrades, { key: data.key, val: data.val() }];
+      setTradesArr(loadedTrades);
+      setMasterTradesArr(loadedTrades);
+    });
+    return () => {
+      unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [user]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => filterTrades(filter), [filter]);
+  useEffect(() => {
+    setTradesArr(sortTrades(tradesArr, sort));
+    setTradelines(
+      tradesArr.map((obj) => (
+        <TradeLineItem
+          key={obj.key}
+          uid={obj.key}
+          val={obj.val}
+          handleDel={handleDel}
+          handleEdit={handleEdit}
+        />
+      ))
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [tradesArr, sort]);
+
+  const sortTrades = (arr, sortMethod) => {
+    const sortedArr = arr.sort(
+      (objA, objB) =>
+        (sortMethod === "timeA" ? -1 : 1) * (objB.val.date - objA.val.date)
+    );
+    return sortedArr;
+  };
+
+  const filterTradesInput = (filterCat) => {
+    setFilterCat(filterCat);
+    setShowModal(true);
+  };
+
+  const filterTrades = (filter) => {
+    const loadedTrades = [];
+    if (filter === "none") {
+      setTradesArr(masterTradesArr);
+    } else {
+      for (let i = 0; i < masterTradesArr.length; i++) {
+        if (masterTradesArr[i].val[filterCat] === filter) {
+          loadedTrades.push(masterTradesArr[i]);
+        }
+      }
+      setTradesArr(loadedTrades);
+    }
+  };
+
+  const handleDel = (tradeUid) => {
+    remove(ref(database, `${user}/${TRADES_KEY}/${tradeUid}`)).then(() => {
+      let newTradesArr = tradesArr;
+      for (let i = 0; i < newTradesArr.length; i++) {
+        if (newTradesArr[i].key === tradeUid) {
+          newTradesArr.splice(i, 1);
```

Comment:
```suggestion
          const newTradesArr = [...tradesArr].splice(i, 1);
```

Reply:
The way you copied newTradesArr from tradesArr, you will actuall perform the splice on the original array as well

Reply:
Pass by reference!